src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState_Checksum.cs

```diff
@@ -14,6 +14,21 @@
 
 namespace Microsoft.CodeAnalysis;
 
+internal readonly struct ProjectCone(ProjectId rootProjectId, IReadOnlySet<ProjectId> projectIds) : IEquatable<ProjectCone>
```

Comment:
Core idea is that we have a strongly-typed view of the cone that we can capture in the scope object.  That cone info can help reduce unnecessary work with syncing.